tests/test_config.py

```diff
@@ -9,310 +9,205 @@
-
-    def test_check_types(self):
+    def test_add_recipe_file(self, core_ini, recipes_ini):
         """
-        Checks to make sure we throw the correct error when an unknown data
-        type is requested
+        Builds a master config from the files, add a recipe file, check that the recipe name is in the list
         """
+        # Build a master config file using multiple files
+        mcfg = MasterConfig(path=core_ini)
+        mcfg.cfg = mcfg.add_files([recipes_ini])
+        assert 'topo_basic_recipe' in [r.name for r in mcfg.recipes]
 
-        # Call out a BS entry type to raise the error
-        checkers = get_checkers()
-        invalids = ['str', 'filepath', 'criticalfile']
-        valids = ['bool', 'criticaldirectory', 'criticalfilename',
-                  'discretionarycriticalfilename', 'datetime',
-                  'datetimeorderedpair', 'directory', 'filename', 'float',
-                  'int', 'string', 'url']
-
-        for z, values in enumerate([invalids, valids]):
-            for kw in values:
-                line = ["type = {}".format(kw), "description = test"]
-                cfg = {"section": {"test": ConfigEntry(name='test',
-                                                       parseable_line=line)}}
-                # invalids
-                if z == 0:
-                    self.assertRaises(ValueError, check_types, cfg, checkers)
-
-                # valids
-                else:
-                    assert check_types(cfg, checkers)
-
-
-if __name__ == '__main__':
-    import sys
-    sys.exit(unittest.main())
+#     @pytest.mark.parametrize("valid_type")
+#     def test_check_types(mcfg_tester):
+#         """
+#         Checks to make sure we throw the correct error when an unknown data
+#         type is requested
+#         """
+#
+#         # Call out a BS entry type to raise the error
+#         checkers = get_checkers()
+#         valids = ['bool', 'criticaldirectory', 'criticalfilename',
+#                   'discretionarycriticalfilename', 'datetime',
+#                   'datetimeorderedpair', 'directory', 'filename', 'float',
+#                   'int', 'string', 'url']
+#
+#         for kw in valids:
+#             line = ["type = {}".format(kw), "description = test"]
+#             cfg = {"section": {"test": ConfigEntry(name='test',
+#                                                    parseable_line=line)}}
+#             assert check_types(cfg, checkers)
+#
+#
+# def test_check_types_exception(mcfg_tester):
+#     # Call out a BS entry type to raise the error
+#     checkers = get_checkers()
+#     invalids = ['str', 'filepath', 'criticalfile']
+#
+#     for kw in invalids:
+#         line = ["type = {}".format(kw), "description = test"]
+#         cfg = {"section": {"test": ConfigEntry(name='test',
+#                                                parseable_line=line)}}
+#         with pytest.raises(ValueError):
+#             check_types(cfg, checkers)
```

Comment:
Curious why these tests are commented out?

Reply:
Hmmm not sure. I remember doing it but I think I was planning to move them but now I see I never got to it. This is the danger of programming with kids in the room lol. I will get a fix in here.

Reply:
Wait for the day when your kids point out the bugs you have in your software 😄